ui/platform_window/whole_screen_move_loop.h

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2013 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef UI_PLATFORM_WINDOW_WHOLE_SCREEN_MOVE_LOOP_H_
+#define UI_PLATFORM_WINDOW_WHOLE_SCREEN_MOVE_LOOP_H_
+
+#include <stdint.h>
+
+#include "base/callback.h"
+#include "base/compiler_specific.h"
+#include "base/macros.h"
+#include "base/memory/weak_ptr.h"
+#include "ui/base/cursor/cursor.h"
+#include "ui/events/platform/platform_event_dispatcher.h"
+#include "ui/gfx/geometry/vector2d_f.h"
+#include "ui/gfx/image/image_skia.h"
+#include "ui/gfx/native_widget_types.h"
+#include "ui/gfx/x/x11_types.h"
+#include "ui/views/widget/desktop_aura/x11_move_loop.h"
+#include "ui/views/widget/desktop_aura/x11_move_loop_delegate.h"
+
+
+
+namespace ui {
+class MouseEvent;
+class ScopedEventDispatcher;
+class XScopedEventSelector;
+class PlatformWindowDelegate;
+
+// Runs a nested run loop and grabs the mouse. This is used to implement
```

Comment:
It's not clear to me how this wires into the larger world. Can you explain?

Reply:
Ok, the WholeScreenMoveLoop does the same stuff as X11WholeScreenMoveLoop except some parts. For example, the events are handled differently, and it's initialized a bit differently. 

That object does the following - it creates an invisible xwindow and maps all the events to it by explicit grab. Wherever a mouse pointer is moved, all the events go to that object, which modifies events to have a root location as their location (this is done for a proper mouse locations update in Env, because TabDragController uses the last mouse pointer events location to know when it can attach the being dragged window to a tab strip). Then the location of the pointer is passed to WindowMoveLoopClient, which updates X11WindowOzone bounds.

Reply:
Once we get release event, run loop is stopped and WindowTree::PerformWindowMove() calls OnChangeCompleted with a result of the move loop.